lib/archethic/utils.ex

```diff
@@ -811,4 +812,17 @@ defmodule Archethic.Utils do
       |> DateTime.from_naive!("Etc/UTC")
     end
   end
+
+  # sobelow_skip ["DOS.StringToAtom"]
+  @doc """
+  Uses String.to_existing_atom() if fails , then it creates a new atom.
+  Prevents String.to_atom DDOS attack.sobelow.
+  """
+  @spec convert_string_to_atom(string :: String.t()) :: atom()
+  def convert_string_to_atom(string) do
+    String.to_existing_atom(string)
+  rescue
+    _ ->
+      String.to_atom(string)
```

Comment:
The goal of the String.to_existing_atom is to avoid atom table exhaustion. 
By catching the throw you might exhaust the atom limits.

Reply:
> The goal of the String.to_existing_atom is to avoid atom table exhaustion. By catching the throw you might exhaust the atom limits.

hi this function , may be ,it was created to mitigate, that if atom does not already exist then  
`String.to_existing_atom ` will throw a error or raise, causing the Process to crash. To avoid crash
if a atom does not exists then create a atom,  `String.to_atom`. So the Process don't crash.
The first time execution of SC will require atom creation. 

`https://github.com/nccgroup/sobelow/blob/master/lib/sobelow/dos/string_to_atom.ex`

```  
@moduledoc """
  # Denial of Service via `String.to_atom`
  In Elixir, atoms are not garbage collected. As such, if user input
  is passed to the `String.to_atom` function, it may result in memory
  exhaustion. Prefer the `String.to_existing_atom` function for untrusted
  user input.
```

Reply:
difficult to conclude , what is expected?

Reply:
I believe this happens because the module where the atom is used is not loaded yet (modules are lazy loaded). The issue might be fixed by ensuring the module is loaded. See that example:

```
iex(27)> String.to_existing_atom("add_uco_transfer")
** (ArgumentError) errors were found at the given arguments:

  * 1st argument: not an already existing atom

    :erlang.binary_to_existing_atom("add_uco_transfer", :utf8)
    iex:27: (file)
```

```
iex(41)> Code.ensure_loaded(Archethic.Contracts.Interpreter.TransactionStatements)
{:module, Archethic.Contracts.Interpreter.TransactionStatements}
```

```
iex(45)> String.to_existing_atom("add_uco_transfer")
:add_uco_transfer
```